### ISSUE TITLE
haploid->diploid HTP screen

### ISSUE THREAD

**Issue by ValWood** [pombase/curation#2991]
https://www.pombase.org/reference/PMID:29259000

should be diploids ? (except the mating phenotype?)

I think there will be quite a few things to migrate to diploids. I can check  'meiosis specific ' phenotypes, but will be good to fix the HTP first. This is the first one.

**Comment by ValWood** [comment]
Also
https://www.pombase.org/reference/PMID:16169489
this will take care of a large portion

**Comment by mah11** [comment]
This will require a way to specify diploid genotypes in a PHAF, which I don't think we have yet. @kimrutherford any suggestions?

There are too many to do manually in Canto (min. 3 clicks per gene to create diploid genotypes, then min. 3 clicks per annotation) - PMID:29259000 has 563 genes with 599 annotations, 157 to FYPO:0000121 and 442 to FYPO:0000151. PMID:16169489 "only" has 153 annotations for 151 genes.

**Comment by ValWood** [comment]
Definitely needs automating. Can discuss on Thursday.

**Comment by kimrutherford** [comment]
> This will require a way to specify diploid genotypes in a PHAF, which I don't think we have yet. @kimrutherford any suggestions?

If the genotypes are diploid but homozygous we can probably handle that we a flag when loading.  If not, it's ... tricky.

**Comment by mah11** [comment]
One of the papers has a mix of haploids and diploids, so the flag would have to go on each line. It wouldn't be much work to add, though; just tell me what to put where.

**Comment by mah11** [comment]
I don't *think* I've seen heterozygous diploids in PHAF-ed papers so far ........

**Comment by kimrutherford** [comment]
> One of the papers has a mix of haploids and diploids, so the flag would have to go on each line. It wouldn't be much work to add, though; just tell me what to put where.

Adding a "ploidiness" column would be a quick fix.  If we add the column at the end with could have "haploid" as the default so you'd only have to change some lines in some file.

I'll try making those changes and see if it would work.  I'll let you know if I make progress.

**Comment by mah11** [comment]
Sounds good. Let me know when you've got it working and I can plug flags in.

**Comment by mah11** [comment]
p.s. "ploidy" is the genetics jargon for what we're talking about

**Comment by ValWood** [comment]
> I don't think I've seen heterozygous diploids in PHAF-ed papers so far ........

PMID_20473289_phaf.tsv	is haploinsufficent and haploproficient (for essential genes).
This might provide a  cross-check. If a paper has a high percentage of essential genes, the mutants might usually be heterozygous diploids, perhaps?

**Comment by mah11** [comment]
> PMID_20473289_phaf.tsv is haploinsufficent and haploproficient (for essential genes).

OK; I didn't do any of the annotation from that one.

> This might provide a cross-check. If a paper has a high percentage of essential genes, the mutants might usually be heterozygous diploids, perhaps

That doesn't sound easier than just looking at the genotypes in the paper ...

**Comment by ValWood** [comment]
;)
https://github.com/pombase/curation/issues/954
(there was a section that I didn't do)

r1854 | mah79 | 2014-06-27 17:56:43 +0100 (Fri, 27 Jun 2014) | 1 line
new phaf for PMID:9003295

**Comment by ValWood** [comment]
Add a column
ploidy value haploid or homozygous diploid,  or heterozygous diploid 
haploid assumed if blank

so if ade1delta/ade1(WT)

**Comment by mah11** [comment]
clarification:

step 1: add `ploidy` column; start with values `haploid` (default) or `homozygous diploid`
step 2 (to do after step 1 is working): extend allowed values to include `heterozygous diploid with WT`; non-WT allele will still be the only one entered in the rest of the row

step 3 (quite possibly a long time from now): replace this hack with better handling for all multi-allele genotypes in HTP files, whether haploid or diploid

**Comment by mah11** [comment]
I've added the column to PHAFs, i.e. step 1 is ready for loading (or testing thereof). I did `homozygous diploid` for PMID:16169489 and haploid or diploid as appropriate for PMID:29259000, and there's a blank column for the rest.

**Comment by ValWood** [comment]
https://www.pombase.org/reference/PMID:28410370 has quite a few.

would it be risky to auto migrate all of these haploids
https://www.pombase.org/term/FYPO:0002739
to homozygous diploid (I'm guessing the heterozygotes would be more likely to be sp[eciofied already)

**Comment by mah11** [comment]
> https://www.pombase.org/reference/PMID:28410370 has quite a few.

Not really; most of the phenotypes from that paper are mating. Also, even the sporulation and ascus phenotypes were determined by starting with homothallic haploids, and letting them divide, switch mating type, mate, and undergo meiosis & sporulation.

> would it be risky to auto-migrate all of these haploids
> https://www.pombase.org/term/FYPO:0002739
> to homozygous diploid

It wouldn't be risk-free, because some of them might have been done by the start-with-haploids method as above (and where does that pat1-114 business fit it? it never fails to confuse me). I'm kinda neutral wrt whether it's better to err one way vs. the other, i.e. neither cheerlead for nor oppose changing them.

> (I'm guessing the heterozygotes would be more likely to be specified already)

Given how recently we got diploids working in Canto, this may not be a safe assumption for papers curated a while ago. But I do think it's safe to say none of us would have curated a heterozygous diploid using a haploid genotype; more likely we wouldn't have done the annotation yet.

**Comment by ValWood** [comment]
so in this case:
Also, even the sporulation and ascus phenotypes were determined by starting with homothallic haploids, and letting them divide, switch mating type, mate, and undergo meiosis & sporulation.

they would be  homozygous diploids when they sporulate, so should we change to diploid?

> It wouldn't be risk-free, because some of them might have been done by the start-with-haploids method as above (and where does that pat1-114 business fit it? it never fails to confuse me).

I always assumed the start with haploids method  and mate a homothallic strain was the only way to create a diploid ? Is there another way?

pat-114 is just for synchronization....




delayed onset of sporulation
abnormal sporulation resulting in formation of ascus with fewer than four spores
abnormal sporulation
large ascus

**Comment by mah11** [comment]
> so in this case:
> 
>> Also, even the sporulation and ascus phenotypes ...
> 
> they would be homozygous diploids when they sporulate, so should we change to diploid?

This seems like a grey area to me. On one hand, you're right, normally only diploids sporulate; on the other hand, the experiments use haploid strains at the outset, and follow everything that happens including transiently forming diploids.

I don't actually oppose changing the genotypes to diploid. I just understand why they could be curated as haploid in the first place.

> I always assumed the start with haploids method and mate a homothallic strain was the only way to create a diploid ? Is there another way?

You can also start with a heterothallic haploid, and mate it with a heterothallic haploid of opposite mating type. 

I don't think I can advise what's best for pombe genotypes; I'm too used to S. cerevisiae where it's a *lot* clearer when to curate haploid or diploid genotypes.

**Comment by ValWood** [comment]
This is where we need @Antonialock !

**Comment by kimrutherford** [comment]
I've made a separate issue about the code change so I don't forget: pombase/pombase-chado#832

**Comment by ValWood** [comment]
I have this ticket on my list butt I don't know if there is anything to discuss?
If so @mah11  could you summarize on the call (if we get to it)

**Comment by ValWood** [comment]
Follow up.
On a recent call I said I would review annotations from the Blythe paper which had a mixture of diploid and haploid alleles. It turned out that this was due to the HTP data being correctly diploid and the LTP being haploid as it was curated pre-Canto diploid curation.

So,  probably some phenotypes will always be diploid...and if we can identify these we can migrate the annotations.

**Comment by ValWood** [comment]
What is to do here. I read the ticket but I git a bit lost.

**Comment by mah11** [comment]
I think ...

- see if Step 2 [here](https://github.com/pombase/curation/issues/2991#issuecomment-830182375) is either done or ready to do;
- decide whether to just punt on the question of "did they start with haploids or diploids" question and just convert all annotations to diploid if the cells had to be diploid somewhere along the line; if so, add "diploid" in affected PHAFs

but I could be missing something else.

**Comment by ValWood** [comment]
So @kimrutherford 

step 2 (to do after step 1 is working): extend allowed values to include heterozygous diploid with WT; non-WT allele will still be the only one entered in the rest of the row

I'm still a bit confused by step3. I guess the question is which htp phenotype datasets might  get  either of the diploid qualifiers?

**Comment by ValWood** [comment]
Already we can do haploid homozygous diploid
1. Add "heterozygous diploid with WT" /heterozygous diploid with delta"(won't do until  we have an example)
2. We will not add "heterozygous diploid" where both alleles are non WT for now (we have no cases of this - we will need to do them in Canto)
3. Figure out on a case-by-cases basis if we can migrate any existing annotations 
to diploid alleles
